bundles/org.eclipse.passage.lic.equinox/src/org/eclipse/passage/lic/internal/equinox/EquinoxPassage.java

```diff
@@ -45,23 +41,32 @@ public boolean canUse(String feature) {
 
 	@Override
 	public void checkLicense(String feature) {
-		// accessManager.executeAccessRestrictions(configuration);
-		throw new UnsupportedOperationException();
+		Optional<Framework> framework = framework();
+		if (!framework.isPresent()) {
+			reportErrorPassageSabotaged();
+		}
+		new Access(framework.get()).check(feature);
 	}
 
-	private Optional<FrameworkSupplier> frameworkSupplier() {
+	private Optional<Framework> framework() {
 		BundleContext context = FrameworkUtil.getBundle(getClass()).getBundleContext();
 		try {
 			return context.getServiceReferences(FrameworkSupplier.class, null).stream() //
 					.map(context::getService) //
 					// DI is used only to get rid of overwhelming dependencies here
 					.filter(supplier -> supplier.getClass().getName()
 							.equals("org.eclipse.passage.seal.internal.demo.DemoFrameworkSupplier")) //$NON-NLS-1$
-					.findAny();
+					.findAny() //
+					.flatMap(FrameworkSupplier::get);
 		} catch (InvalidSyntaxException e) {
 			log.error(EquinoxMessages.EquinoxPassage_no_framework, e);
 			return Optional.empty();
 		}
 	}
 
+	// FIXME: invoke executors here?
```

Comment:
Yes, it would be better I think, we can try it in the next PRs

Reply:
Yes, executors are evolving not in scope of [565656](https://bugs.eclipse.org/bugs/show_bug.cgi?id=565656) task.